ros2topic/ros2topic/api/__init__.py

```diff
@@ -206,50 +206,56 @@ def qos_profile_from_short_keys(
     return profile
 
 
-def add_qos_arguments(parser: ArgumentParser, subscribe_or_publish: str, default_profile_str):
+def add_qos_arguments(
+    parser: ArgumentParser, entity_type: str,
+    default_profile_str: str = 'default', extra_message: str = ''
+) -> None:
     parser.add_argument(
         '--qos-profile',
         choices=rclpy.qos.QoSPresetProfiles.short_keys(),
         help=(
-            f'Quality of service preset profile to {subscribe_or_publish} with'
+            f'Quality of service preset profile to {entity_type} with'
             f' (default: {default_profile_str})'),
         default=default_profile_str)
     default_profile = rclpy.qos.QoSPresetProfiles.get_from_short_key(default_profile_str)
     parser.add_argument(
         '--qos-depth', metavar='N', type=int,
         help=(
-            f'Queue size setting to {subscribe_or_publish} with '
-            '(overrides depth value of --qos-profile option)'))
+            f'Queue size setting to {entity_type} with '
+            '(overrides depth value of --qos-profile option, default: '
+            f'{default_profile.depth})'))
     parser.add_argument(
         '--qos-history',
         choices=rclpy.qos.QoSHistoryPolicy.short_keys(),
         help=(
-            f'History of samples setting to {subscribe_or_publish} with '
+            f'History of samples setting to {entity_type} with '
             '(overrides history value of --qos-profile option, default: '
             f'{default_profile.history.short_key})'))
     parser.add_argument(
         '--qos-reliability',
         choices=rclpy.qos.QoSReliabilityPolicy.short_keys(),
         help=(
-            f'Quality of service reliability setting to {subscribe_or_publish} with '
+            f'Quality of service reliability setting to {entity_type} with '
             '(overrides reliability value of --qos-profile option, default: '
-            'Compatible profile with running endpoints )'))
```

Comment:
i think this message for `ros2 topic pub` is wrong description, because only `ros2 topic echo` tries to find the compatible profile with concerned publishers when it starts. in this function, i think it always print the default values for qos parameters.